GArDen/convert/molecule.py

```diff
@@ -0,0 +1,326 @@
+'''
+interface between networkx graphs and the world of chemistry via rdkit
+
+functionality:
+# gernerating networkx graphs:
+sdf_to_nx(file.sdf)
+smi_to_nx(file.smi)
+smiles_to_nx(smilesstringlist)
+
+# graph out:
+draw(nx)
+nx_to_smi(graphlist, path_to_file.smi)
+
+
+#bonus: garden style transformer.
+class MoleculeToGraph
+'''
+from sklearn.base import BaseEstimator, TransformerMixin
+from rdkit import Chem
+from rdkit.Chem import AllChem
+from rdkit.Chem import Draw
+from IPython.core.display import display
+import networkx as nx
+import eden.graph as edengraphtools
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class MoleculeToGraph(BaseEstimator, TransformerMixin):
+    def __init__(self, file_format='sdf'):
+        """Constructor.
+
+        valid 'file_format' strings and what the transformer will expect
+        smiles: list of smiles strings
+        smi: path to .smi file
+        sdf: pat to .sdf file
+        """
+        self.file_format = file_format
+
+    def transform(self, data):
+        """Transform."""
+        try:
+            if self.file_format == 'smiles':
+                graphs = smiles_to_nx(data)
+            elif self.file_format == 'smi':
+                graphs = smi_to_nx(data)
+            elif self.file_format == 'sdf':
+                graphs = sdf_to_nx(data)
+            else:
+                raise Exception('file_format must be smiles smi or sdf')
+            for graph in graphs:
+                yield graph
+
+        except Exception as e:
+            logger.debug('Failed iteration. Reason: %s' % e)
+            logger.debug('Exception', exc_info=True)
+
+
+def set_coordinates(chemlist):
+    for m in chemlist:
+        if m:
+            # updateprops fixes "RuntimeError: Pre-condition Violation"
+            m.UpdatePropertyCache(strict=False)
+            tmp = AllChem.Compute2DCoords(m)
+        else:
+            print '''set coordinates failed..'''
+
+
+def draw(graphs, n_graphs_per_line=5, size=250, title_key=None, titles=None, smiles=False):
+    # we want a list of graphs
+    if isinstance(graphs, nx.Graph):
+        print "give me a list of graphs"
+
+    # make molecule objects
+    chem = [nx_to_rdkit(graph) for graph in graphs]
+    # print chem
+    # calculate coordinates:
+    set_coordinates(chem)
+
+    # take care of the subtitle of each graph
+    if title_key:
+        legend = [g.graph[title_key] for g in graphs]
+    elif titles:
+        legend = titles
+    else:
+        legend = [str(i) for i in range(len(graphs))]
+
+    if smiles == False:
+        # make the image
+        image = Draw.MolsToGridImage(chem, molsPerRow=n_graphs_per_line, subImgSize=(size, size), legends=legend)
+        # display on the spot
+        display(image)
+    else:
+        for m in chem:
+            print Chem.MolToSmiles(m)
```

Comment:
Removed the pring part.
I am not sure though if that was your only concern since imo the whole print 
stuff should move.